src/Forms/Main.cs

```diff
@@ -3875,9 +3875,13 @@ private void TextBoxSourceClick(object sender, EventArgs e)
             ShowSourceLineNumber();
         }
 
-        private void TextBoxSourceKeyDown(object sender, KeyEventArgs e)
+        private void textBoxSource_KeyUp(object sender, KeyEventArgs e)
         {
             ShowSourceLineNumber();
+        }
+
+        private void TextBoxSourceKeyDown(object sender, KeyEventArgs e)

```

Comment:
Since the caret will only change position after the process is out of this method so ShowSourceLineNumber() will be called and **textBoxSource.GetLineFromCharIndex(textBoxSource.SelectionStart)** will return line of caret's current position.